ecl/hql/hqlattr.cpp

```diff
@@ -3776,6 +3776,71 @@ ITypeInfo * setStreamedAttr(ITypeInfo * _type, bool setValue)
     }
 }
 
+//---------------------------------------------------------------------------------
+double queryLikelihood(IHqlExpression * expr)
+{
+    IInterface * match = meta.queryExistingProperty(expr, EPlikelihood);
+    if (match)
+    {
+        IHqlExpression * likelihoodExpr = static_cast<IHqlExpression *>(match);
+        IValue * value = likelihoodExpr->queryChild(0)->queryValue();
+        return value->getRealValue();
+    }
+
+    double likelihoodValue = -1.0;
```

Comment:
Efficiency.  I would move the declaration of likelihoodExpr above the case statement, and only set it from the value if it is non null